provisioning/service/src/Contract/ContractApiHttp.cs

```diff
@@ -46,25 +46,32 @@ internal class ContractApiHttp : IContractApiHttp
         public ContractApiHttp(
             Uri baseAddress,
             IAuthorizationHeaderProvider authenticationHeaderProvider,
-            ProvisioningServiceHttpSettings httpSettings)
+            ProvisioningServiceClientOptions options)
         {
             _baseAddress = baseAddress;
             _authenticationHeaderProvider = authenticationHeaderProvider;
 
+            if (options.HttpClient != null)
+            {
+                _httpClientObj = options.HttpClient;
+                _httpClientHandler = null;
```

Comment:
What does assigning null do here? Was there a previous instance? Did it need to be disposed?

Reply:
It doesn't do anything since this is the default value. I can remove this line